### ISSUE TITLE
Proposal: refactoring src/Tag.php

### ISSUE THREAD

**Issue by machour** [mpdf/mpdf#551]
Hi,

I would like to help refactoring the huge `src/Tag.php` file.
I first started with a [basic cleanup](https://github.com/mpdf/mpdf/compare/development...machour:tag-code-cleanup?expand=1), but as I became more familiar with the file, I think that the code needs to be completely split up in order to become more maintainable.

I'm thinking of creating a new `Tag/` folder, and have a class per Tag, i.e:
```shell 
Tag
├── AbstractTag.php
├── Blockquote.php
├── Span.php
├── Table.php
└── Tr.php
```

`AbstractTag` would define two abstracts methods `open()` and `close()`, along with some helpers methods.

As this is a huge piece of work, I'd like to get approval from mPDF maintainers before diving in head first 😄 

Let me know if it's okay with you, or if you have other plans I'm unaware of for this file.

**Comment by finwe** [comment]
This sounds awesome! I had a similar thoughts about this, so by all means go ahead.

**Comment by machour** [comment]
Closed by #565 🕺
